modules/partition-replicator/src/test/java/org/apache/ignite/internal/partition/replicator/ZoneResourcesManagerTest.java

```diff
@@ -0,0 +1,94 @@
+ */
+
+package org.apache.ignite.internal.partition.replicator;
+
+import static org.apache.ignite.internal.testframework.IgniteTestUtils.bypassingThreadAssertions;
+import static org.apache.ignite.internal.testframework.matchers.CompletableFutureMatcher.willCompleteSuccessfully;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.when;
+
+import java.nio.file.Path;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import org.apache.ignite.internal.catalog.descriptors.CatalogZoneDescriptor;
+import org.apache.ignite.internal.components.LogSyncer;
+import org.apache.ignite.internal.manager.ComponentContext;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.apache.ignite.internal.testframework.ExecutorServiceExtension;
+import org.apache.ignite.internal.testframework.InjectExecutorService;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.internal.testframework.WorkDirectoryExtension;
+import org.apache.ignite.internal.tx.storage.state.TxStatePartitionStorage;
+import org.apache.ignite.internal.tx.storage.state.rocksdb.TxStateRocksDbSharedStorage;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@ExtendWith(WorkDirectoryExtension.class)
+@ExtendWith(ExecutorServiceExtension.class)
+class ZoneResourcesManagerTest extends BaseIgniteAbstractTest {
+    private TxStateRocksDbSharedStorage sharedStorage;
+
+    @Mock
+    private CatalogZoneDescriptor zoneDescriptor;
+
+    @Mock
+    private LogSyncer logSyncer;
+
+    private ZoneResourcesManager manager;
+
+    @WorkDirectory
+    private Path workDir;
+
+    @InjectExecutorService
+    private ScheduledExecutorService scheduler;
+
+    @InjectExecutorService
+    private ExecutorService executor;
+
+    @BeforeEach
+    void init() {
+        sharedStorage = new TxStateRocksDbSharedStorage(workDir, scheduler, executor, logSyncer, () -> 0);
+
+        manager = new ZoneResourcesManager(sharedStorage);
+
+        when(zoneDescriptor.partitions()).thenReturn(10);
+
+        assertThat(sharedStorage.startAsync(new ComponentContext()), willCompleteSuccessfully());
+    }
+
+    @AfterEach
+    void cleanup() {
+        assertThat(sharedStorage.stopAsync(), willCompleteSuccessfully());
+    }
+
+    @Test
+    void createsTxStatePartitionStorage() {
+        TxStatePartitionStorage txStatePartitionStorage = bypassingThreadAssertions(
+                () -> manager.getOrCreatePartitionStorage(zoneDescriptor, 1)
+        );
+
+        assertThat(txStatePartitionStorage, is(notNullValue()));
+    }
```

Comment:
I'd also add a test for ZoneResourcesManager#close that will ensure that all storages for several zones that were created will be closed. Similar to what you've added in ZonePartitionReplicaImplTest.